### ISSUE TITLE
[SDK52] expo-doctor fails on default template even though the right version is installed

### ISSUE THREAD

**Issue by moritzlang** [expo/expo#32393]
### Minimal reproducible example

https://github.com/moritzlang/sdk-52-expo-doctor-failing

### What platform(s) does this occur on?

Android, iOS, Web

### Where did you reproduce the issue?

in Expo Go

### Summary

I used the default template:
`npx create-expo-app --template default@beta`

and installed `expo-image`:
`npx expo install expo-image`

This installs `expo-image` with a version of  `~2.0.0-preview.0`

Now when I run `npx expo-doctor@latest` it fails with this error:
```
The following packages should be updated for best compatibility with the installed expo version:
  expo-image@2.0.0-preview.0 - expected version: ~2.0.0-preview.0
Your project may not work correctly until you install the expected versions of the packages.
Found outdated dependencies
```

Which seems weird since the correct version is installed?

Thanks for your help!

### Environment

```text
expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.5
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.20.3 - ~/.nvm/versions/node/v18.20.3/bin/node
      Yarn: 1.22.19 - /opt/homebrew/bin/yarn
      npm: 10.7.0 - ~/.nvm/versions/node/v18.20.3/bin/npm
    Managers:
      CocoaPods: 1.12.1 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.5, iOS 17.5, macOS 14.5, tvOS 17.5, visionOS 1.2, watchOS 10.5
    IDEs:
      Android Studio: 2022.3 AI-223.8836.35.2231.10811636
      Xcode: 15.4/15F31d - /usr/bin/xcodebuild
    npmPackages:
      expo: ~52.0.0-preview.5 => 52.0.0-preview.5 
      expo-router: ~4.0.0-preview.3 => 4.0.0-preview.3 
      react: 18.3.1 => 18.3.1 
      react-dom: 18.3.1 => 18.3.1 
      react-native: 0.76.0 => 0.76.0 
      react-native-web: ~0.19.13 => 0.19.13 
    npmGlobalPackages:
      eas-cli: 10.0.2
    Expo Workflow: managed
```

### Expo Doctor Diagnostics

```text
The following packages should be updated for best compatibility with the installed expo version:
  expo-image@2.0.0-preview.0 - expected version: ~2.0.0-preview.0
Your project may not work correctly until you install the expected versions of the packages.
Found outdated dependencies
Advice: Use 'npx expo install --check' to review and upgrade your dependencies.

One or more checks failed, indicating possible issues with the project.
```

**Comment by entiendoNull** [comment]
+1, I've also run into this on every SDK52 project I've started now.

**Comment by keith-kurak** [comment]
Hi all! Thanks for reporting. We have a PR in-flight to fix this here: https://github.com/expo/expo/pull/32160

**Comment by the-simian** [comment]
I see there's a PR. 
In case others end up here: I'm experiencing this with
```
The following packages should be updated for best compatibility with the installed expo version:
  expo-dev-client@5.0.0-preview.2 - expected version: ~5.0.0-preview.2
  expo-router@4.0.0-preview.4 - expected version: ~4.0.0-preview.4
  @expo/metro-runtime@4.0.0-preview.0 - expected version: ~4.0.0-preview.0
```
Look forward to the fix

**Comment by the-simian** [comment]
I can verify the push 2 hours ago here: https://github.com/expo/expo/pull/32160
has totally resolved this for me

Thank you!!

**Comment by entiendoNull** [comment]
> Hi all! Thanks for reporting. We have a PR in-flight to fix this here: #32160

Seems to have done the trick! Works well now 👍
